Civi/RemoteEvent/CompilerPass.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Civi\RemoteEvent;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+use CRM_RemoteEvent_ExtensionUtil as E;
+
+class CompilerPass implements CompilerPassInterface {
+
+  public function process(ContainerBuilder $container) {
+    if ($container->hasDefinition('action_provider')) {
+      $actionProviderDefinition = $container->getDefinition('action_provider');
+      $actionProviderDefinition->addMethodCall('addAction',
```

Comment:
This would be slightly better style:
```php
$actionProviderDefinition->addMethodCall('addAction', [
  'RemoteEventSpawnEvent',
  'Civi\RemoteEvent\Actions\SpawnEvent',
  E::ts('RemoteEvent spawn'),
  [],
]);
```

Instead of `'Civi\RemoteEvent\Actions\SpawnEvent'` the `::class` notation could be used: `SpawnEvent::class`.

Reply:
I would prefer to use this way because it aligns more to the documentation of the extension:
https://lab.civicrm.org/extensions/action-provider/-/blob/master/docs/howto_create_an_action_in_an_extension.md